backend/src/main/java/ai/verta/modeldb/versioning/DatasetComponentDAORdbImpl.java

```diff
@@ -425,4 +428,107 @@ private Folder getFolder(Session session, String commitSha, String folderSha) th
     throw new ModelDBException(
         "Unexpected logic issue found when fetching blobs", Status.Code.UNKNOWN);
   }
+
+  private InternalFolderElementEntity getFolderElement(
```

Comment:
```suggestion
  /**
   * get the Folder Element pointed to by the parentFolderHash and elementName 
   * @param session
   * @param parentFolderHash : folder hash of the parent
   * @param elementName : element name of the element to be fetched
   * @return
   */ 
  private InternalFolderElementEntity getFolderElement(
``` Please write basic comments for functions